prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -594,31 +587,70 @@ class ReportFunction : Logging {
                 return sb.toString()
             }
 
-            fun writeConsolidatedArray(field: String, array: List<ResultDetail>) {
-                val rowsByGroupingId = hashMapOf<String, MutableList<Int>>()
-                val messageByGroupingId = hashMapOf<String, String>()
+            fun createPropertiesByGroupingId(array: List<ResultDetail>): GroupedProperties {
+                val itemsByGroupingId = mutableMapOf<String, MutableList<Int>>()
+                val messageByGroupingId = mutableMapOf<String, String>()
+                val scopesByGroupingId = mutableMapOf<String, String>()
+                val itemDetailsWithGroupingId = mutableMapOf<String, MutableList<Int>>()
                 array.forEach { resultDetail ->
                     val groupingId = resultDetail.responseMessage.groupingId()
-                    if (!rowsByGroupingId.containsKey(groupingId)) {
-                        rowsByGroupingId[groupingId] = mutableListOf()
+                    if (!itemsByGroupingId.containsKey(groupingId)) {
+                        itemsByGroupingId[groupingId] = mutableListOf()
                         messageByGroupingId[groupingId] = resultDetail.responseMessage.detailMsg()
+                        scopesByGroupingId[groupingId] = resultDetail.scope.toString()
```

Comment:
Yeah. The work Sean did grouped them by error, so when you're looking at the verbose output, you'll see the following JSON:
```
    "scope" : "ITEM",
    "message" : "Invalid date: 's' for element 'Date_result_released' ('date_result_released'). Reformat to yyyyMMdd.",
    "itemNums" : "Rows: 15, 19 to 20",
    "itemDetails" : [ {
      "itemNum" : "15",
      "groupingId" : "'Date_result_released' ('date_result_released')s"
    }, {
      "itemNum" : "19",
      "groupingId" : "'Date_result_released' ('date_result_released')s"
    }, {
      "itemNum" : "20",
      "groupingId" : "'Date_result_released' ('date_result_released')s"
    } ]
```
Since the `groupingId` is based on the error itself, if the invalid date is, say, `10-20-2021`, all rows in `itemNums` will be of `scope: ITEM`